### ISSUE TITLE
Doxygen produce latex that makes pdflatex run out of memory 

### ISSUE THREAD

**Issue by ngothan** [doxygen/doxygen#8049]
Latex output of Doxygen causes pdflatex to run out of memory. Using doxygen-1.8.15 running on fedora 32

**To Reproduce**
on Fedora 32:
sudo dnf install git doxygen graphviz dblatex texlive-latex texlive-makeindex
git clone --depth=1 git@github.com:gcc-mirror/gcc.git
mkdir build
cd build
../gcc/configure --enable-languages=c,c++ --disable-gcc --disable-lib{atomic,cc1,gcc,cpp,ffi,gomp,itm,sanitizer,ssp,vtv}
eval `grep '^target=' config.log`
make configure-target
make -C $target/libstdc++-v3 doc-pdf-doxygen

**Describe the bug**
dflatex fails and the log file shows:

Runaway text?
\@ifundefined {etoctocstyle}{\let \etoc@startlocaltoc \@gobble \let \etoc@setto
cdepth \ETC.
! TeX capacity exceeded, sorry [main memory size=5000000].
<inserted text> ...6.2.28}\contentsline {para\ETC.

l.139 \begin{document}
                      ^^M
If you really absolutely need more capacity,
you can ask a wizard to enlarge me.

Here is how much of TeX's memory you used:
 16738 strings out of 494553
 238522 string characters out of 6177319
 5000001 words of memory out of 5000000
 20192 multiletter control sequences out of 15000+600000
 6245 words of font info for 17 fonts, out of 8000000 for 9000
 14 hyphenation exceptions out of 8191
 53i,1n,92p,7517b,98s stack positions out of 5000i,500n,10000p,200000b,80000s
!  ==> Fatal error occurred, no output PDF file produced!

**Expected behavior**
PDF file is produced.

**Additional context**
This works fine with doxygen-1.8.14 and fails with 1.8.15 and new vesions. I also tested the lastest doxygen (1.9.0) from master branch. It still has this issue. I attached the diff of the latex generated by doxygen-1.8.14 and 1.8.15 so you can see.

[doxygen-latex.diff.gz](https://github.com/doxygen/doxygen/files/5262051/doxygen-latex.diff.gz)

**Comment by albert-github** [comment]
I used the procedure as described by you on Cygwin with doxygen 1.8.18 (the default installed version installed version on my Cygwin) and I got a PDF file and no error message.

At the end of the log file I have:
```
Here is how much of TeX's memory you used:
 86162 strings out of 480617
 3310321 string characters out of 5910439
 753257 words of memory out of 5000000
 58756 multiletter control sequences out of 15000+600000
 590221 words of font info for 108 fonts, out of 8000000 for 9000
 1141 hyphenation exceptions out of 8191
 55i,17n,96p,5268b,19014s stack positions out of 5000i,500n,10000p,200000b,80000s
{/usr/share/texmf-dist/fonts/enc/dvips/base/8r.enc}</usr/share/texmf-dist/fon
ts/type1/public/amsfonts/cm/cmex10.pfb></usr/share/texmf-dist/fonts/type1/publi
c/amsfonts/cm/cmmi10.pfb></usr/share/texmf-dist/fonts/type1/public/amsfonts/cm/
cmmi12.pfb></usr/share/texmf-dist/fonts/type1/public/amsfonts/cm/cmmi5.pfb></us
r/share/texmf-dist/fonts/type1/public/amsfonts/cm/cmmi7.pfb></usr/share/texmf-d
ist/fonts/type1/public/amsfonts/cm/cmmi8.pfb></usr/share/texmf-dist/fonts/type1
/public/amsfonts/cm/cmr10.pfb></usr/share/texmf-dist/fonts/type1/public/amsfont
s/cm/cmr5.pfb></usr/share/texmf-dist/fonts/type1/public/amsfonts/cm/cmr7.pfb></
usr/share/texmf-dist/fonts/type1/public/amsfonts/cm/cmsy10.pfb></usr/share/texm
f-dist/fonts/type1/public/amsfonts/cm/cmsy5.pfb></usr/share/texmf-dist/fonts/ty
pe1/public/amsfonts/cm/cmsy6.pfb></usr/share/texmf-dist/fonts/type1/public/amsf
onts/cm/cmsy7.pfb></usr/share/texmf-dist/fonts/type1/public/amsfonts/cm/cmsy8.p
fb></usr/share/texmf-dist/fonts/type1/urw/courier/ucrb8a.pfb></usr/share/texmf-
dist/fonts/type1/urw/courier/ucrr8a.pfb></usr/share/texmf-dist/fonts/type1/urw/
courier/ucrro8a.pfb></usr/share/texmf-dist/fonts/type1/urw/helvetic/uhvb8a.pfb>
</usr/share/texmf-dist/fonts/type1/urw/helvetic/uhvb8ac.pfb></usr/share/texmf-d
ist/fonts/type1/urw/helvetic/uhvbo8a.pfb></usr/share/texmf-dist/fonts/type1/urw
/helvetic/uhvr8a.pfb></usr/share/texmf-dist/fonts/type1/urw/helvetic/uhvro8a.pf
b>
Output written on refman.pdf (3149 pages, 16030658 bytes).
PDF statistics:
 103139 PDF objects out of 114338 (max. 8388607)
 97199 compressed objects within 972 object streams
 39036 named destinations out of 45951 (max. 500000)
 2476 words of extra memory for PDF output out of 10000 (max. 10000000)
```

I'm using: `This is pdfTeX, Version 3.14159265-2.6-1.40.21 (TeX Live 2020/Cygwin) (preloaded format=pdflatex 2020.6.10)
git clone https://github.com/gcc-mirror/gcc.git`

Which version of LaTeX are you using?

Note I had to clone by means of:
```
git clone https://github.com/gcc-mirror/gcc.git
```
furthermore I created the build directory `build` directly inside the `gcc` tree as `gcc/build` so for the config step I used:
```
../configure --enable-languages=c,c++ --disable-gcc --disable-lib{atomic,cc1,gcc,cpp,ffi,gomp,itm,sanitizer,ssp,vtv}
```

- My main concern is the LaTeX version and the used packages, which one are you using?
- When going in `<target>/libstdc++-v3/doc/doxygen/latex` what happens when you comment out in `refman.tex` all the lines between:
  ```
  %--- Begin generated contents ---
  ```
  and
  ```
  %--- End generated contents ---
  ```
  and run in this directory `pdflatex refman.tex`?

**Comment by jwakely** [comment]
This still fails for me on Fedora 37

```
pdflatex refman
This is pdfTeX, Version 3.141592653-2.6-1.40.22 (TeX Live 2021) (preloaded format=pdflatex)
 restricted \write18 enabled.
entering extended mode
(./refman.tex
LaTeX2e <2020-10-01> patch level 4
L3 programming layer <2021-05-07>
make[2]: [Makefile:17: refman.pdf] Error 1 (ignored)
make[2]: Leaving directory '/tmp/test/build/x86_64-pc-linux-gnu/libstdc++-v3/doc/doxygen/latex'
... error
! TeX capacity exceeded, sorry [pool size=5905152].
make[1]: *** [Makefile:969: stamp-pdf-doxygen] Error 12
```

**Comment by albert-github** [comment]
@jwakely 

- Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.
- Please also specify the **full** doxygen version used (`doxygen -v`).

**Comment by jwakely** [comment]
That's using Doxygen 1.9.6 (875558cea412de2535b0390c4625bcd0f6bdb411) or Doxygen 1.9.5 (same result with both).

I can't attach a small example, it won't run out of memory with a small example!

I can attach a tarball of the entire gcc repo, but that will be huge. The steps above still reproduce it.

**Comment by jwakely** [comment]
Using `git clone --depth=1 --branch=releases/gcc-13.1.0 git@github.com:gcc-mirror/gcc.git` will be persistent, that is a tag for today's GCC 13.1 release, which will not change.

**Comment by jwakely** [comment]
So here are the full steps to reproduce it with a specific, persistent version of the GCC code:

```
git clone --depth=1 --branch=releases/gcc-13.1.0 https://github.com/gcc-mirror/gcc.git
mkdir build
cd build
../gcc/configure --enable-languages=c,c++ --disable-gcc --disable-lib{atomic,cc1,gcc,cpp,ffi,gomp,itm,sanitizer,ssp,vtv}
eval `grep '^target=' config.log`
make configure-target
make -C $target/libstdc++-v3 doc-pdf-doxygen
```

**Comment by albert-github** [comment]
Indeed the entire gcc repo would not make sense.
Problem is the
```
TeX capacity exceeded, sorry [pool size=5905152]
```
this is something doxygen cannot do a lot about / can do nothing about, it is a memory requirement oft the Tex engine and one has to set the memory / pool size of the Tex engine.
(see e.g. https://tex.stackexchange.com/questions/109568/how-to-increase-pool-size-in-miktex-2-9)

**Comment by jwakely** [comment]
But I think there's something wrong with the doxygen output which causes TeX to run out of memory. Increasing the memory doesn't help, it consumes all available memory, however high the limit is.

Using `lualatex` instead of `pdflatex` causes it to spin at 100% CPU for more than 30 minutes (then I killed it as it's obviously not going to finish).

**Comment by jwakely** [comment]
> * When going in `<target>/libstdc++-v3/doc/doxygen/latex` what happens when you comment out in `refman.tex` all the lines between:

I tried this. There are more than 2600 lines between there. The `pdflatex refman.tex` command finishes successfully, in 2.5s.

**Comment by albert-github** [comment]
Very large projects are indeed sometimes a problem, we need to investigate where exactly there is a problem and how this can be solved (won't be easy).

**Comment by jwakely** [comment]
I'm trying to bisect the `\input` files in the .tex file to find out at which point it runs out of memory, to see if there's a specific file that's causing it.

**Comment by jwakely** [comment]
As a workaround, `SOURCE_BROWSER=NO` makes the build succeed. It seems that the large source files are a major contributor to the problem.

It might be that this is just too large for TeX to handle, and we should just stop trying.

**Comment by albert-github** [comment]
I don't think that just setting `SOURCE_BROWSER=NO` solves the general problem, I think that in this particular case it sees to it that the "pool size" is under the set limit.
The major question, for me, is what is recorded in the pool and  how can we see the contents of this "pool".

**Comment by JBThiel** [comment]
AbiWord uses Doxygen, and I saw this out of memory error, same as mentioned above:
`! TeX capacity exceeded, sorry [main memory size=5000000].
`
I was attempting to document it, but my debugging session got interrupted, then a couple days later I couldn't reproduce it.  Possibly intermittent, may relate to VERBATIM_HEADERS, ...  I was bisecting against various combinations of flags under Doxywizard.Expert.SourceBrowser.

AbiWord uses Doxygen extensively and would make a good test case for you.  Currently the Latex output also has many problems when you 'make' in the Doxygen-created latex subdir, in regards underfull \hbox, possible also vbox, problems with Unicode chars, etc.  You have to manually hit Return many times, to make the Latex job finish.  A full run produces an 18000+ page document.

It would be nice for Doxygen to emit some write flags or macros to make Latex auto-continue thru the errors.  And make sure it is not generating underfull boxes, check for Unicode chars, etc.

This is with AbiWord from https://github.com/AbiWord/abiword
commit = 6940426 2023-03-08,
on Doxygen 1.12.0
on Gentoo Linux

**Comment by albert-github** [comment]
@JBThiel 

You wrote:
> AbiWord uses Doxygen extensively

Where can I find the resulting documentation of the doxygen process on Abiword?

I see a `Doxyfile` though here the `GENERATE_LATEX = NO` so no latex generated, also the `Doxyfile` is from doxygen 1.7.5 and has not been updated for 14 years.
The project itself has not been updated for 2 years (so looks like a bit abandoned, I also found Abiword on https://gitlab.gnome.org/World/AbiWord with last update 6 month ago).

I tried with doxygen 1.12.o (and doxygen 1.13.2 and with the current master) but when generating the pdf file I got something like:
```

[6056]
! Missing # inserted in alignment preamble.
<to be read again>
                   \cr
l.332 \end{longtabu}
```

so I don't even get to the point you mentioned (or maybe I got already a bit further?) ...

- How did you run doxygen on Abiword? and subsequently tried to generate the pdf file?

**Comment by JBThiel** [comment]
Yes, the pre-built Doxyfile seemed rather outdated.
Running Doxywizard on the Doxyfile, it shows more GUI elements,
and it seems like the new/changed elements are in RED.

The included Doxyfile was set to HTML default, and I switched it to Latex.
Presumably all the backends are supposed to be
equally well supported as output from Doxygen (?).

The Latex material is being generated in docs/dox/latex.
After the Doxyfile run, I went in that subdir and type 'make'.
You have to hit return a couple dozen times at various snag points, but the latex job will continue
and then it actually finished (when it didn't get the memory error).
It makes a DVI file, which you can convert via 'dvipdf refman.dvi'.

This is with the TEXLIVE 2024 package on Gentoo Linux.
which reports 'latex' is
  This is pdfTeX, Version 3.141592653-2.6-1.40.26 (TeX Live 2024 Gentoo Linux) (preloaded format=latex)

Overall process for latex pdf:
                   
1) run Doxywizard on the Doxyfile;
adjust options under Wizard.Output, unselect HTML, select Latex.
I think I left it "as intermediate format for Postscript", just as a first run attempt;
the other settings about PDF might be better.

{UPDATE: I tested now also "as intermediae format for PDF",
  and this makes refman.pdf directly, not the dvi, so dvipdf step below is not needed}

{UPDATE: with the setting "as intermediate format for hyperlinked PDF" I get a new different failure,
and the latex run aborts as follows. You cannot continue here and the PDF file is damaged, not openable:}

```
[10033]
...
Overfull \hbox (84.43492pt too wide) in paragraph at lines 97--98
[]\T1/phv/m/n/10 Referenced by [], [], [],

Overfull \hbox (151.4134pt too wide) in paragraph at lines 97--98
[]\T1/phv/m/n/10 , [], [], [],

Overfull \hbox (77.80354pt too wide) in paragraph at lines 97--98
[]\T1/phv/m/n/10 , [], [], [],
! Unable to read an entire line---bufsize=200000.
Please increase buf_size in texmf.cnf.

```
2) adjust Doxywizard other options under Expert.SourceBrowser.
  It seems like if either SOURCE_BROWSER or VERBATIM_HEADERS is on,
  then it's pulling in other stuff, makes the final doc bigger.
  (18000+ pages, versus I think it was 16000 or 17000 pages without).
  I have been running the tests with SOURCE_BROWSER = OFF, but VERBATIM_HEADERS = ON.
  ( and all of STRIP_CODE_COMMENTS, REFERENCED_BY_RELATION, REFERENCES_RELATION
    and REFERENCES_LINK_SOURCE  = ON )
2 options are in RED in the GUI here:
  REFERENCED_BY_RELATION
  REFERENCES_RELATION
I'm guessing Red because these are new options in the GUI than in the old Doxyfile.

3) Doxywizard.Run.RunDoxygen button.  Makes the whole docs/dox/latex tree.

4) cd docs/dox/latex, and 'make'

5) I get the same error you showed, as the first one
```
! Missing # inserted in alignment preamble.
<to be read again>
                   \cr
l.332 \end{longtabu}
```

If you just hit RETURN here, Latex can recover and continue.
It will pause at a couple dozen more such problems, so just keep hitting Return.
Eventually it finishes at about 18000 pages.
This is obviously not ideal, because you have to keep interacting with the process.
Though more facile with Latex in the past, I don't use it much directly like this now,
so I don't recall if there are option flags or commands you can insert in the preamble
for example, to make it auto-continue or ignore such errors.

6) Now the refman.dvi is made, and can be converted to PDF via
  dvipdf refman.dvi
On Gentoo, this dvipdf is part of package
  app-text/ghostscript-gpl

{UPDATE: If you give "as intermediate format for PDF"  option mentioned above,
then latex makes a pdf directly, so you don't need this step }

I'm fairly new to Doxygen, and evaluating it for documenting large codebases.
Especially to make a single file PDF.
I found AbiWord had the pre-built Doxyfile, and it looks like
they have special Doxygen-format comments in the code, and the full run makes a giant file.
So it seems like a good test case.

Though as you mention it is perhaps somewhat outdated now.
On the other hand, it would be good for Doxygen to maintain strong back-compatibility,
so it can use older Doxyfiles and comments.  It looks like good attention is already being given to
that idea, as with the RED marked options in the GUI.

I had found this bug report, because I initially encountered and was researching
the out-of-memory error.  But then it didn't reproduce.  There are a lot of Doxywizard options,
and the necessity of hitting Return a bunch of times adds confusion. Maybe I hit some stray
keys input at the Latex prompt for example, instead of just Return.
  Also I am not sure if 'make clean' is necessary in the docs/dox/latex subdir
between each Doxywizard Generation run, and if I was doing that every time.

So I wrote it up anyway in regards the issues about underful boxes, Unicode, need to hit Return, etc.
And that AbiWord has the heavy Doxygen use, good for testing.

Basically, everything the Latex run is printing (which is a lot), besides the [nnnn] page numbers is a problem.
It's mostly underfull hboxes. So Doxygen could make sure not to create such underfulls, or as
I recall there are latex/tex codes to make it be more lenient about what it
considers underfull, so it won't warn about them as much.

If preferable, we could open some new issues about those other issues.
If I see the memory error again, I will try to get an exact reproduction sequence.

**Comment by albert-github** [comment]
Thanks for the extensive report.
I'm first going to concentrate on the the:
```
! Missing # inserted in alignment preamble.
<to be read again>
                   \cr
l.332 \end{longtabu}
```
error (it is not good that one has to hit the return key as this means there might be a, better said is, a problem in the generated file). When looking at the HTML output (with a "lint checker") I also see the the HTML output is not 100% correct.
The problem seems to be that there is an empty table in the output (as I found some lonely `<table>` tags), doxygen does emit a warning for it but still should create a valid table or better, as the table is empty, not place it in the output at all.

> resumably all the backends are supposed to be
equally well supported as output from Doxygen (?).


Not completely, we try to keep the different output formats up to date, but we are also limited to the knowledge we have and the reports that are submitted. The best maintained formats are HTML, XML and LeTeX.


Regarding the generation of the latex output, I would not use `LATEX_CMD_NAME         = latex` as command but set it to `pdflatex` or leave it empty (for the majority of output languages it will choose `pdflatex` and for a few to `xelatex`).


- could you post the `Doxyfile` you used or post the results from the doxywizard from the "run" panel when you set `condensed and hit the `show configuration` button (so we are talking about the exact setup).

**Comment by JBThiel** [comment]
On this Gentoo system, /usr/bin/{latex,pdflatex} are both symlinks to /usr/bin/pdftex.
So 'latex' should be running same thing as pdflatex, presumably.
I will just leave that setting for now for consistency to prior test runs, unless it seems to be a problem.

That is a great feature about Condensed Show Config.
Below is the condensed config with the Latex suboption set to "as intermediate format for PDF".
This is from loaded the original AbiWord Doxyfile, plus set to Latex, and whatever updates
it made itself from the newer version, plus whatever I have been clicking around the last few days.
I have kept the Doxywizard GUI open.

Good find on the tables.
The main things that require Return are those tables (a batch of 4 Returns, then another 4)
and a bunch of bad Unicode or UTF-8 chars as below.

The huge number of underfull or overfull boxes, warnings about references not found, etc. 
don't wait for Return, they are just warnings flying by, but it's a huge flow of stuff which
will hide any more important warnings (also I think Latex might put some markers in the margins for bad boxes).
It would be better if the Latex run was eventually a nice clean log of [nnnn] pages .


```
# Doxyfile 1.12.0

PROJECT_NAME           = AbiWord
OUTPUT_DIRECTORY       = docs/dox
ABBREVIATE_BRIEF       =
FULL_PATH_NAMES        = NO
JAVADOC_AUTOBRIEF      = YES
INHERIT_DOCS           = NO
EXTRACT_ALL            = YES
EXTRACT_PRIVATE        = YES
EXTRACT_STATIC         = YES
CASE_SENSE_NAMES       = YES
INPUT                  = .
FILE_PATTERNS          = *.cpp \
                         *.h \
                         *.TXT \
                         *.mm \
                         *.c \
                         *.t.*
RECURSIVE              = YES
EXCLUDE                = hello
EXAMPLE_PATTERNS       =
REFERENCED_BY_RELATION = YES
REFERENCES_RELATION    = YES
IGNORE_PREFIX          = UT_ \
                         XAP_ \
                         AP_ \
                         ap_ \
                         AD_ \
                         GR_ \
                         EV_ \
                         ev_ \
                         TF_ \
                         FL_ \
                         fl_ \
                         FV_ \
                         fv_ \
                         PT_ \
                         pf_ \
                         PO_ \
                         PP_ \
                         pp_ \
                         AV_ \
                         XAV_ \
                         fb_ \
                         fd_ \
                         fg_ \
                         IE_ \
                         ie_
GENERATE_HTML          = NO
MATHJAX_RELPATH        = http://www.mathjax.org/mathjax
SEARCHENGINE           = NO
LATEX_CMD_NAME         = latex
PDF_HYPERLINKS         = NO
ALLEXTERNALS           = YES

```

These are the other 3 kinds of Unicode/UTF-8 Latex errors that require Return to continue, bunches of these various ones:


```
[9183]) (./ie__imp__ClarisWorks_8h.tex
Underfull \hbox (badness 10000) in paragraph at lines 3--5
) (./ie__imp__ClarisWorks_8h_source.tex
! LaTeX Error: Invalid UTF-8 byte sequence (�re).
See the LaTeX manual or LaTeX Companion for explanation.
Type  H <return>  for immediate help.
 ...
l.9 ...*\ Copyright\ (C)\ 2000\ Hubert\ Figui�re}}




[9548]) (./ie__impexp__KWord__1_8h.tex) (./ie__impexp__KWord__1_8h_source.tex
[9549]
! LaTeX Error: Invalid UTF-8 byte "B7.
See the LaTeX manual or LaTeX Companion for explanation.
Type  H <return>  for immediate help.
 ...
l.22 ...n\ Street,\ Fifth\ Floor,\ Boston,\ MA��}}




[9693]) (./ODi__ListLevelStyleFormats_8h.tex
! LaTeX Error: Unicode character أ (U+0623)
               not set up for use with LaTeX.
See the LaTeX manual or LaTeX Companion for explanation.
Type  H <return>  for immediate help.
 ...
l.6 ...\+\_\+\+LISTLEVELSTYLE\+\_\+\+ARABIC}~"{}أ
                                                  , ب, ت, ..."{}

```

**Comment by albert-github** [comment]
> On this Gentoo system, /usr/bin/{latex,pdflatex} are both symlinks to /usr/bin/pdftex.

Not becessarily, when I remeber correctly by checking `argv[0]` for the real used command it is possible to do something different.

Regarding the error type:
```
l.9 ...*\ Copyright\ (C)\ 2000\ Hubert\ Figui�re}}
```
this looks like an `INPUT_ENCODING` problem, looks like the `INPUT_ENCODING = ISO-8859-1` might work ...

**Comment by JBThiel** [comment]
* I set under Expert.Input.INPUT_ENCODING, change from UTF-8, to  ISO-8859-1
This WORKED, it eliminated all the Return pauses in the latex run caused by the UTF8/Unicode.
So the only pauses are from the table issue.

Looks like this encoding fix should be sent upstream to AbiWord, if it hasn't made it
to the latest version you found.
It should be set in their Doxyfile, per their source files have that alternate encoding.
So it is not a Doxygen/Latex error; the error messages from Latex are correct.
Maybe I will send them a mention eventually, referring to this report.

Note, that is some quite detailed setting, not easily discoverable.
The INPUT_ENCODING is just a plain text field, you can type anything.
At least it has a good help msg, which references libiconv, 
but 'iconv -l' lists 100s of encodings, the user has to really know exactly what their encoding is.

Maybe the INPUT_ENCODING field could offer some popup list of the most common encodings,
or the help msg give some guidance about this.  Also seems a good candidate
for a Doxygen FAQ.


* I changed Expert.Latex.LATEX_CMD_NAME from 'latex' to empty.
This causes the latex/Makefile to have LATEX_CMD?=pdflatex
The Latex run works apparently the same, although the output file was not identical size, 
so there must be some slight difference.
(size change only about 150 bytes, in 31 MB files - might just be the engine name or such)

**Comment by albert-github** [comment]
Regarding the empty table I've just pushed a proposed patch, pull request #11580

**Comment by albert-github** [comment]
The `INPUT_ENCODING` is indeed a difficult problem and quite a lot of users are not aware of the encoding they are using.
A more complicating thing is that different versions of `iconv` might support a different set of conversions, so it is hard for doxygen to determine what might be possible. Also the doxywizard drop down box currently does not support the possibility to enter your own values (and seen the way things are, automatically, generated it would be rather quite a bit of work to implement it).

The way I normally determine the file encoding is by using the *nix `file` command and match this against the list of `iconv -l`.
(On windows a bit more complex as you must have the commands from e.g. Cygwin or Windows Subsystem for Linux, see e.g. https://learn.microsoft.com/en-us/windows/wsl/faq)

**Comment by JBThiel** [comment]
Nice work on the table fix.

For INPUT_ENCODING, I think just giving some guidance in the help message would be a big help,
and works with the current text field. So it doesn't have to be UI mod. Something like, where it says
  Default is UTF-8,

ADD:
Other common encodings are:
Linux/OSX/Unix = ...
Windows/Dos = ...
Other = ...
See 'iconv -l' for a full list.

And list the top 6 or 10 encodings.
Also up in the main paragraph, some mention like:
"Try adjusting this if LaTeX generates UTF8/Unicode character errors."

**Comment by albert-github** [comment]
The fix regarding the incorrect `<table>` tag has been integrated into the master version of doxygen.
